packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -138,6 +139,24 @@ class OverlayEntry implements Listenable {
     _overlay!._didChangeEntryOpacity();
   }
 
+  /// Whether the content of this [OverlayEntry] can be used to size the
+  /// [Overlay].
+  ///
+  /// In most situations the Overlay sizes itself based on its incoming
+  /// constraints to be as large as possible. However, if that would result in
+  /// an infinite size, it has to rely on one of its children to size itself. In
+  /// this situation, the Overlay will consult the topmost [OverlayEntry] that
+  /// has this property set to true, lay it out with unconstrained
+  /// [BoxConstraints], and force all other OverlayEntries to have the same
+  /// size.
+  ///
+  /// OverlayEntries that set this to true must be able to handle unconstrained
+  /// [BoxConstraints].
+  ///
+  /// Setting this to true has no effect if the OverlayEntry uses a [Positioned]
+  /// widget to position itself in the Overlay.
+  final bool canSizeOverlay;
```

Comment:
nit: I'd drop the "can" and just start with "sizes". The word "can" instills doubt. I'm guessing you added it because there's no guarantee that this overlay is the top-most one and so it will not necessarily be used to determine the size of the view. However, that's just a precedence rule or a tie-breaker. It's still OK to say that it sizes the view.

Reply:
> The word "can" instills doubt.

That's exactly the doubt I want to install because in the vast majority of cases (the Overlay is getting non-infinite constraints) it does not size the overlay. The cases where it does size it are very, very, very rare. I want people to see this property, pause, and read the documentation to learn when it will size the overlay.